Remove automatic config generation (#2820)
jwilm/alacritty: CHANGELOG.md

### Packaging

- Minimum Rust version has been bumped to 1.36.0

> - Automatic config generation was removed, please ship alacritty.yml in /usr/share/docs

Author: I'm not sure about this comment though

Reviewer: I'm also not sure about this.

- Arch Linux's `community/alacritty` is currently putting things in `/usr/share/doc/alacritty/example/alacritty.yml`.
- I'd be curious to hear @ChrisMacNaughton's thoughs, as the maintainer of our snapcraft
- macOS is another story since we're mainly distributed as a "cask" which is just a `.app` moved into the Applications folder. We don't (unless I'm mistaken) get the opportunity to install anything. Alacritty itself could check for the template and install it when opened, but I don't like this approach at all.

Our own `INSTALL.md` may want to be updated as well. It currently refers to installing things like the icons, and man pages. This seems like a natural addition there.

Author: The key point is not to install config. It's about obtaining the *right* default one, since packagers can ship whatever version they want, so they should also ship a config for this specific version somewhere now. We're just acknowledge them that they can't rely on automatic config generation for this purpose anymore. So adding instructions to INSTALL.md is kind of strange, since we're already have one in README, however I'm not opposed to it and would like to hear more opinions on this.

Reviewer: @kchibisov how do you propose macOS packagers to solve this problem? I'm not a fan of the only method to find the default config requiring the internet (e.g. checking GitHub).

Author: homebrew can ship docs. I've proposed adding a flag for that purposes, but @chrisduerr wasn't happy about it iirc. We can create alacritty-config(5) man page for this purposes, but it's a not cross platform solution and a maintenance burden.

> I'm not a fan of the only method to find the default config requiring the internet (e.g. checking GitHub).

You need to download Alacritty anyway, it's not shipped as a default terminal on macOS/Windows atm.

Author: I think the message in CHANGELOG is misleading atm, but I've already stated it. Maybe `Automatic config generation was removed, please ship alacritty.yml for easy documentation obtain` will make more sense.

Reviewer: > You need to download Alacritty anyway, it's not shipped as a default terminal on macOS/Windows atm.

Thing is, one might download and install once, but then configure later. Ideally we make the offline experience as good as possible.

Author: flag for alacritty to generate default config manually

Author: So `alacritty --generate-config`

Reviewer: I've looked into `brew cask` a bit, and unless I'm missing something it's not really designed to support this use case. But... we could hack the `postflight` hook to install the configuration:

```rb
# ... rest of cask description

postflight do
  path_to_config = ...  # From DMG or separate download?
  prefix = `/usr/local/bin/brew --prefix`.chomp
  doc = "#{prefix}/share/doc/alacritty"
  FileUtils.mkdir_p(doc)
  # TODO: Should really be in the Cellar and symlinked to here.
  FileUtils.cp(path_to_config, "#{doc}/alacritty.yml"
end
```
In a normal brew formula (not a cask) this could be as simple as `doc.install 'alacritty.yml'` but this doesn't seem to work here.

Reviewer 2: I don't think it's reasonable to ship our default config file on Windows or macOS to be honest. No matter where we put it, people are much, much more likely to find it on github than somewhere on their system.

> - Automatic config generation was removed, please ship alacritty.yml for easy documentation obtain

Reviewer 2: ```suggestion
- Config is not generated anymore, please consider distributing the alacritty.yml as documentation
```
